R/check.r

```diff
@@ -77,6 +79,11 @@ check <- function(pkg = ".",
     warning("`cleanup` is deprecated", call. = FALSE)
   }
 
+  if (missing(error_on) && !interactive()) {
```

Comment:
I am glad to pick up this pattern: how to reveal default args in signature and use `match.arg()`, while not necessarily defaulting to the first element.

Reply:
FWIW I generally think you _should_ default to the first argument, just because that is the normal convention. In this particular case I wanted to have different default behavior when run non-interactively, so needed to do this.

Maybe I should have used `error_on = NULL` so could have had the logic only in one place rather than two, I am not sure...

Reply:
yeah, think about the kittens before you use `missing()` :)

Reply:
The exact thing just came up in usethis re: incrementing versions. It make sense to have "major", "minor", "patch", "dev", in that order. But in the absence of explicit info ... you probably don't want to increment the major version (in fact, we don't increment at all in that case). Also, I like revealing the possible values in the signature vs. documentation.